src/components/HistoricalForecast/HistoricalForecast.tsx

```diff
@@ -1,6 +1,69 @@
+import { useState } from 'react';
+// import Button from '@mui/material/Button';
+import Field from '@mui/material';
+import Grid from '@mui/material/Grid';
+import { Formik, Form } from 'formik';
+import * as yup from 'yup';
 import DateRangePicker from 'components/DateRangePicker';
+import Textfield from 'components/FormsUI/Textfield';
+import SelectWrapper from 'components/FormsUI/Select';
+import WeatherWidget from 'components/WeatherWidget';
+import Location from 'types/location';
 import * as classes from './styles';
 
+const INITIAL_FORM_VALUES = {
+  language: '',
+  startDate: Date,
+  endDate: Date,
+  location: '',
+  temperatureUnit: '',
+};
+
+const VALIDATION_SCHEME = yup.object().shape({
+  language: yup.string(),
+  startDate: yup.date().required('Required'),
+  endDate: yup.date().required('Required'),
+  location: yup.string().required('Required'),
+  temperatureUnit: yup.string(),
```

Comment:
Let's read initial values of `language` and `temperatureUnit` from context.
But it should be validated more strictly - it should be one of values
When a form is submitted these values should be updated in context.